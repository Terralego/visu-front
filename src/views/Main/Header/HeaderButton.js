import React from 'react';
import PropTypes from 'prop-types';

import {
  Button,
  Classes,
} from '@blueprintjs/core';
import classnames from 'classnames';
import './styles.scss';

export const HeaderButton = ({ id, icon, iconPath, alt }) => (
  <Button
    className={`${Classes.MINIMAL} header-button ${id}`}
    minimal
    id={id}
  >
    <img src={iconPath} alt={alt} className={classnames({ icon })} />
  </Button>
);

HeaderButton.propTypes = {
  id: PropTypes.string.isRequired,
  icon: PropTypes.string,
  iconPath: PropTypes.string,
  alt: PropTypes.string,
};

HeaderButton.defaultProps = {
  icon: 'icon',
  iconPath: '',
  alt: '',
};

export default HeaderButton;
