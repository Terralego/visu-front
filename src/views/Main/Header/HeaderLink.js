import React from 'react';
import { NavLink } from 'react-router-dom';

export const HeaderLink = ({ href, children, ...props }) => {
  if (!href) return <span {...props}>{children}</span>;

  if (href.match(/^http/)) return <a href={href} {...props}>{children}</a>;

  return (
    <NavLink to={href} {...props}>{children}</NavLink>
  );
};

export default HeaderLink;
