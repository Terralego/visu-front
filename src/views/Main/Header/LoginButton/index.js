import compose from '@terralego/core/utils/compose';
import withDeviceSize from '@terralego/core/hoc/withDeviceSize';
import { withTranslation } from 'react-i18next';
import { connectAuthProvider } from '@terralego/core/modules/Auth';

import LoginButton from './LoginButton';

export default compose(
  withDeviceSize(),
  withTranslation(),
  connectAuthProvider('authenticated', 'logoutAction'),
)(LoginButton);
