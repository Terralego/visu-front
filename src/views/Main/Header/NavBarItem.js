import React from 'react';
import {
  Position,
  Tooltip,
} from '@blueprintjs/core';

import withEnv from '../../../config/withEnv';
import HeaderLink from './HeaderLink';
import HeaderButton from './HeaderButton';

export const NavBarItem = ({ id, content, iconPath, env: { VIEW_ROOT_PATH }, ...item }) => (
  <Tooltip
    content={item.label}
    position={Position.RIGHT}
    usePortal={false}
  >
    <HeaderLink
      href={(item.href || '').replace('{{VIEW_ROOT_PATH}}', VIEW_ROOT_PATH)}
      onClick={item.onClick}
      data-link-id={id}
    >
      <HeaderButton
        id={id}
        iconPath={iconPath}
        alt={item.label}
        icon={item.icon}
      >
        {item.content}
      </HeaderButton>
    </HeaderLink>
  </Tooltip>
);

export default withEnv(NavBarItem);
