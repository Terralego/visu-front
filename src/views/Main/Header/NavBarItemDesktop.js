import React from 'react';
import PropTypes from 'prop-types';

import {
  Position,
  Tooltip,
} from '@blueprintjs/core';

import withEnv from '../../../config/withEnv';
import HeaderLink from './HeaderLink';
import HeaderButton from './HeaderButton';

export const NavBarItemDesktop = ({
  id,
  content,
  iconPath,
  classNameIcon,
  env: { VIEW_ROOT_PATH },
  ...item
}) => (
  <Tooltip
    content={item.label}
    position={Position.RIGHT}
    usePortal={false}
  >
    <HeaderLink
      href={(item.href || '').replace('{{VIEW_ROOT_PATH}}', VIEW_ROOT_PATH)}
      onClick={item.onClick}
      data-link-id={id}
    >
      <HeaderButton
        id={id}
        iconPath={iconPath}
        alt={item.label}
        classNameIcon={classNameIcon}
      >
        {item.content}
      </HeaderButton>
    </HeaderLink>
  </Tooltip>
);

NavBarItemDesktop.propTypes = {
  env: PropTypes.shape({
    VIEW_ROOT_PATH: PropTypes.string,
  }).isRequired,
  id: PropTypes.string.isRequired,
  iconPath: PropTypes.string,
  item: PropTypes.shape({
    href: PropTypes.string,
    label: PropTypes.string,
  }),
};

NavBarItemDesktop.defaultProps = {
  iconPath: '',
  item: {
    href: '',
    label: '',
    onClick () {},
  },
};

export default withEnv(NavBarItemDesktop);
