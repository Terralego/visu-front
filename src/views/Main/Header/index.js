import React from 'react';
import {
  Navbar,
  NavbarGroup,
} from '@blueprintjs/core';

import NavBarItem from './NavBarItem';
import logo from '../../../images/logo-auran.svg';
import population from '../../../images/population.png';
import habitat from '../../../images/habitat.png';
import mobilite from '../../../images/mobilite.png';
import economie from '../../../images/economie.png';

import './styles.scss';

const navHeader = [
  [
    {
      id: 'welcome',
      label: 'Accueil',
      href: '/',
      iconPath: logo,
    },
    {
      id: 'nav-population',
      label: 'Population',
      href: '/{{VIEW_ROOT_PATH}}/population',
      iconPath: population,
    },
    {
      id: 'nav-habitat',
      label: 'Habitat',
      href: '/{{VIEW_ROOT_PATH}}/habitat',
      iconPath: habitat,
    },
    {
      id: 'nav-mobilite',
      label: 'Mobilité',
      href: '/{{VIEW_ROOT_PATH}}/mobilite',
      iconPath: mobilite,
    },
    {
      id: 'nav-economie',
      label: 'Économie',
      href: '/{{VIEW_ROOT_PATH}}/economie',
      iconPath: economie,
    },
  ], [],
];

/* eslint-disable react/no-array-index-key */
export const Header = () => (
  <div className="main__header">
    <Navbar className="navBar bp3-dark">
      {
        navHeader.map((group, index) => (
          <NavbarGroup key={index} className="navBar__group">
            {group.map(({ component: Component = NavBarItem, ...item }) => (
              <Component
                key={item.label}
                {...item}
              />
            ))}
          </NavbarGroup>
        ))
      }
    </Navbar>
  </div>
);

export default Header;
