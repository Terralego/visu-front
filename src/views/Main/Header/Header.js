import React, { useState, useEffect } from 'react';
import {
  Navbar,
  NavbarGroup,
} from '@blueprintjs/core';
import classnames from 'classnames';
import withDeviceSize from '@terralego/core/hoc/withDeviceSize';

import { fetchAllViews } from '../../../services/visualizer';

import NavBarItemDesktop from './NavBarItemDesktop';
import NavBarItemTablet from './NavBarItemTablet';
import PartnerButton from './PartnerButton';
import logo from '../../../images/terravisu-logo.svg';

import infoSign from '../../../images/info-sign.svg';

import './styles.scss';

const navItemsBase = [
  [
    {
      id: 'welcome',
      label: 'Accueil',
      href: '/',
      iconPath: logo,
      icon: null,
    },
  ], [],
  [{
    id: 'nav-partenaires',
    label: 'Mentions lÃ©gales',
    iconPath: infoSign,
    component: PartnerButton,
  }],

];

/* eslint-disable react/no-array-index-key */
export const Header = ({
  isHeaderOpen = false,
  toggleHeader = () => { },
  isMobileSized,
}) => {
  const [navItems, setNavItems] = useState([]);
  const containerRef = React.useRef();

  useEffect(() => {
    let isUnmounted = false;

    const loadViewList = async () => {
      /** Load the view list and add it to base menu only if component mounted */
      if (isUnmounted) return;
      const views = await fetchAllViews();
      const newNavItems = JSON.parse(JSON.stringify(navItemsBase));
      newNavItems[0] = newNavItems[0].concat(views);
      setNavItems(newNavItems);
    };

    loadViewList();

    return () => {
      isUnmounted = true;
    };
  }, []);


  useEffect(() => {
    const listener = ({ target }) => {
      if (containerRef.current && containerRef.current.contains(target)) {
        toggleHeader();
        return;
      }
      toggleHeader(false);
    };

    document.body.addEventListener('click', listener);

    return () => {
      document.body.removeEventListener('click', listener);
    };
  }, [toggleHeader]);


  return (
    <div
      className={classnames(
        'main__header',
        { 'main__header--mobile': isMobileSized },
        { 'main__header--mobile--open': isMobileSized && isHeaderOpen },
      )}
      ref={containerRef}
    >
      <Navbar className="navBar bp3-dark">
        {
          navItems.map((group, index) => (
            <NavbarGroup key={index} className="navBar__group">
              {group.map(({ component: Component = isMobileSized
                ? NavBarItemTablet : NavBarItemDesktop, ...item }) => (
                  <Component
                    key={item.label}
                    {...item}
                  />
              ))}
            </NavbarGroup>
          ))
        }
      </Navbar>
      {isMobileSized && !isHeaderOpen
        && <div ref={containerRef} className="main__header__target" />
      }
    </div>
  );
};

export default withDeviceSize()(Header);
