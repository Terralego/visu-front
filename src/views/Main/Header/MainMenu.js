import React from 'react';
import {
  Navbar,
  NavbarGroup,
} from '@blueprintjs/core';
import withDeviceSize from '@terralego/core/hoc/withDeviceSize';
import { connectAuthProvider } from '@terralego/core/modules/Auth';
import PropTypes from 'prop-types';


import NavBarItemDesktop from './NavBarItemDesktop';
import NavBarItemTablet from './NavBarItemTablet';

import './styles.scss';

/* eslint-disable react/no-array-index-key */
export const MainMenu = ({
  authenticated,
  isMobileSized,
  navItems,
}) => (
  <Navbar className="navBar bp3-dark">
    {
          navItems.map((group, index) => (
            <NavbarGroup key={index} className="navBar__group">
              {group.map(({ component: Component = isMobileSized
                ? NavBarItemTablet : NavBarItemDesktop, ...item }) => (
                  <Component
                    key={item.label}
                    iconPath={item.iconPath}
                    {...item}
                    authenticated={authenticated}
                  />
              ))}
            </NavbarGroup>
          ))
        }
  </Navbar>
);

MainMenu.propTypes = {
  isMobileSized: PropTypes.bool,
  authenticated: PropTypes.bool,
  navItems: PropTypes.arrayOf(),
};

MainMenu.defaultProps = {
  isMobileSized: false,
  navItems: [],
  authenticated: false,
};

export default withDeviceSize()(connectAuthProvider('authenticated', 'logoutAction')(MainMenu));
