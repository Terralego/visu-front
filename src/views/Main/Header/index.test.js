import React from 'react';
import renderer from 'react-test-renderer';

import { Header } from './Header';

jest.mock('react-router-dom', () => (
  {
    Route: () => null,
    Switch: () => null,
    Link: () => null,
    withRouter: ({ children }) => children,
  }
));

jest.mock('./PartnerButton', () => () => 'PartnerButton');

jest.mock('../../../services/visualizer', () => (
  {
    fetchAllViews: jest.fn(() => [
      {
        name: 'foo',
        slug: 'foo',
        custom_icon: 'logo',
      }, {
        name: 'bar',
        slug: 'bar',
      },
    ]),
  }
));

it('should render correctly', () => {
  const tree = renderer.create(
    <Header />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render correctly with header open in mobile view', () => {
  const tree = renderer.create(
    <Header
      isMobileSized
      isHeaderOpen
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render correctly with header close in mobile view', () => {
  const tree = renderer.create(
    <Header
      location={{ pathname: '/' }}
      env={{ VIEW_ROOT_PATH: '' }}
      settings={{}}
      isMobileSized
      isHeaderOpen={false}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

/* it('should open navbar in mobile', () => {
  const toggleHeader = jest.fn();
  const wrapper = shallow(<Header toggleHeader={toggleHeader} />);
  const instance = wrapper.instance();
  const target = '<div class="main__header__target--test" />';

  instance.containerRef.current = {
    target,
    contains: () => true,
  };
  instance.listener({ target });

  expect(instance.props.toggleHeader).toHaveBeenCalledWith();
}); */
