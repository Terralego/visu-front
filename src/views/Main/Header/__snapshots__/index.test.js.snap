// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="main__header"
>
  <div
    className="bp3-navbar navBar bp3-dark"
  />
</div>
`;
